upgradability/set_code_hash/voting/lib.rs

```diff
@@ -0,0 +1,178 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang as ink;
+
+#[ink::contract]
+mod voting {
+    use ink_env::set_code_hash;
+    use ink_storage::{traits::SpreadAllocate, Mapping};
+    use scale::{Decode, Encode};
+
+    #[derive(Debug, PartialEq, Eq, Encode, Decode)]
```

Comment:
you may want to familiarize yourself with https://www.notion.so/cardinalcryptography/Basic-traits-derivation-and-field-access-for-public-structs-0aa22c4887024547b613db5754e0664a and adjust your derives to it :)